### PR TITLE
Publish Button test in Jest

### PR DESCRIPTION
**Pull request by TarunRaveesh**
Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `grunt jasmine`
* [ ] code is in uniquely-named feature branch and has no merge conflicts
* [ ] PR is descriptively titled
* [ ] PR body includes `fixes #0000`-style reference to original issue #
* [ ] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. 

Thanks!

fixes https://github.com/publiclab/PublicLab.Editor/issues/745

### CODE REVIEW

File: test/ui-testing/publishButton.test.js [hunk 1]
```diff
@@ -6,9 +6,14 @@ beforeAll(async () => {
 });
 
 describe('Publish button', () => {
```
Comment (jywarren) [inline test/ui-testing/publishButton.test.js:8]: OK, building from this - https://stackoverflow.com/questions/45948843/how-to-require-jquery-plugin-in-jest-test-file i'll try a simple fix. If it works, we may want to move that all into a shared config file so all Jest tests can access it. 

```suggestion
global.window = window
global.$ = require('jquery');

describe('Publish button', () => {
```

File: test/ui-testing/publishButton.test.js [hunk 2]
```diff
@@ -5,10 +5,18 @@ beforeAll(async () => {
   await page.goto('file://' + path, {waitUntil: 'domcontentloaded'});
 });
 
+global.window = window
```
Comment (jywarren) [inline test/ui-testing/publishButton.test.js:8]: Hmm now I see: `    ReferenceError: window is not defined` so will try removing this line?

Comment (jywarren) [reply test/ui-testing/publishButton.test.js:8]:
```suggestion
# global.window = window
```

**Comment by jywarren** [comment]
OK - more closely following the example in https://stackoverflow.com/a/48763451 now! Fingers 🤞

**Comment by jywarren** [comment]
OK, so the issue here is that we're trying to use jQuery, but within Puppeteer's browser, we can't directly access jQuery. We'd have to do it in a block like this:

```js
   await page.goto('http://localhost/jest/table.html');
   await page.addScriptTag({url: 'https://code.jquery.com/jquery-3.3.1.slim.min.js'});
   const result = await page.evaluate(() => {
       try {
           var table = $("table").attr("id");
           return table;
       } catch (e) {
           return e.message;
       }
   });
   await console.log("result", result);
```

That's a lot of trouble, and we can't do assertions inside, i think? not sure. But maybe we can install this: https://github.com/melbourne2991/jquery-puppeteer and it would "just work"? worth a shot. I'll try this tomorrow. 

I'm so sorry this is much more complex than we'd thought! We'll get through it together though!

References:

- https://stackoverflow.com/questions/49192329/how-do-i-use-jquery-with-pages-on-puppeteer
- https://dev.to/napolux/how-to-scrap-that-web-page-with-nodejs-and-puppeteer-811
- https://jestjs.io/docs/tutorial-jquery

**Comment by jywarren** [comment]
Or easier maybe native puppeteer selectors?

https://github.com/puppeteer/puppeteer/blob/v10.4.0/docs/api.md#pageselector

File: test/ui-testing/publishButton.test.js [hunk 3]
```diff
@@ -3,17 +3,18 @@ const fs = require('fs');
...
-        $('.ple-module-title input').val('Title');
-        $('.ple-module-title input').keydown();
-        // Check final state of Publish button.
-        expect($('.ple-publish').prop('disabled')).toBe(false);
-    }, timeout);
-
-  });
+  test('publish button gets enabled', () => {
+    // Check initially that Publish button is disabled.
+    expect(page.$('.ple-publish').getAttribute('disabled')).toBe(true);
```
Comment (jywarren) [inline test/ui-testing/publishButton.test.js:13]: `TypeError: page.$(...).getAttribute is not a function`

**Comment by jywarren** [comment]
https://stackoverflow.com/questions/56467696/get-the-value-of-html-attributes-using-puppeteer#56467778

**Comment by TildaDares** [comment]
Hi @TarunRaveesh, we're currently working on converting jasmine tests to jest and the conversion is a bit tricky. @NARUDESIGNS has resolved this issue in #785. You did a great job and we appreciate the time you spent working on this PR. If you'd like to work on another issue, leave a comment here and I'll find another one for you. Thank you!

**Pull request closed without merging**